### PR TITLE
[Enhancement]Creating multiple drop ship purchase orders in one go for sales order with multiple items assigned to different suppliers 

### PR DESCRIPTION
**Pull request by szufisher**
Refer to issue https://github.com/frappe/erpnext/issues/15031

**Business case**

- Sales order with multi items assigned to different suppliers

**As Is**

- by clicking the create purchase order custom menu item, there is popup window to force user select one supplier, system auto fetch all the data for the new drop ship purchase order and route to the new purchase order screen, waiting for user to further modify the order and click SAVE to create the final draft PO, for each supplier, the above mentioned steps need to be repeated

- if draft purchase orders already created for the supplier, click the create purchase order menu item and select the supplier again, system will create the purchase order again without any error message, only when trying to submit the purchase order, there is limit crossed error!

**To Be**

- Allow user to leave the supplier field empty, system creates multi draft drop ship purchase orders for all suppliers assigned to sales order items
- only the supplier by which the corresponding drop ship purchase orders(draft and submitted) not yet created will be available in the supplier drop down list
leave supplier empty
Fig1: Allow user leave the supplier field empty
dropshiporders
Fig2: after multi purchase orders created, route to the list instead of the form view

Please read the pull request checklist to make sure your changes are merged: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

### CODE REVIEW

**Comment by szufisher** [comment]
seems the build failure: testing on accounting and hr modules , is not related to my change, I have no idea how to solve them, can this change still be merged?

File: erpnext/selling/doctype/sales_order/sales_order.py [hunk 1]
```diff
@@ -737,41 +737,61 @@ def update_item(source, target, source_parent):
...
+					"condition": lambda doc: doc.ordered_qty < doc.qty and doc.supplier == supplier
+				}
+			}, target_doc, set_missing_values)
+			if not for_supplier:
+				doc.insert()
+		else:
+			suppliers =[]
+	if suppliers:
+		if not for_supplier:
+			frappe.db.commit()
```
Comment (nabinhait) [inline erpnext/selling/doctype/sales_order/sales_order.py]: Is commit required?

**Comment by nabinhait** [comment]
@szufisher We have fixed some errors in Travis. Can you please rebase and push again?

**Comment by nabinhait** [comment]
merged from console, closing the PR

**Pull request closed without merging**